### PR TITLE
Introduce GATT Client service

### PR DESCRIPTION
**Pull request by KKopyscinski**

### CODE REVIEW

**Comment by KKopyscinski** [comment]
@sjanc @mkasenberg I marked this as ready for review, I think this could be used as base for later, only thing that is left is possible changes that were added in meanwhile in https://github.com/intel/auto-pts/pull/620 and implementetion of some WIDs.

File: wid/gatt_client.py [hunk 1]
```diff
@@ -0,0 +1,1261 @@
...
+    Description: Verify that the Implementation Under Test (IUT) can send
+    Discover characteristic descriptors in database."
+    """
+    MMI.reset()
+    MMI.parse_description(params.description)
+
+    stack = get_stack()
+    stack.gatt_cl.wait_for_descs()
+
+    print(stack.gatt_cl.chrcs)
```
Comment (mkasenberg) [inline wid/gatt_client.py:481]: debug leftovers

**Review by michal-narajowski — CHANGES_REQUESTED** [review]

File: ptsprojects/stack.py [hunk 2]
```diff
@@ -951,13 +951,146 @@ def wait_notification_ev(self, timeout=None):
...
+    def is_prim_disc_complete(self, args):
+        if self.prim_svcs_cnt is not None:
+            if self.prim_svcs_cnt > 0:
+                rc = len(self.prim_svcs) == self.prim_svcs_cnt
+                if rc is True:
+                    self.prim_svcs_cnt = None
+                return rc
+            self.prim_svcs_cnt = None
+            return True
+        return False
```
[the remark this answers is not in the document]
Comment (KKopyscinski) [reply ptsprojects/stack.py]: Added commit that simplifies this

**Comment by michal-narajowski** [comment]
Rebase please.

**Comment by KKopyscinski** [comment]
> Rebase please.

done

**Review by michal-narajowski — APPROVED** [review]

**Pull request merged**